src/solver/optimisation/adequacy_patch_csr/solve_problem.cpp

```diff
@@ -0,0 +1,203 @@
+{
+    int Var;
+    double* pt;
+    for (Var = 0; Var < ProblemeAResoudre->NombreDeVariables; Var++)
+    {
+        pt = ProblemeAResoudre->AdresseOuPlacerLaValeurDesVariablesOptimisees[Var];
+        if (pt)
+        {
+            *pt = ProblemeAResoudre->X[Var];
+            setToZeroIfBelowThreshold(pt, Var, hourlyCsrProblem);
+        }
+        logs.debug() << "[CSR]" << Var << " = " << ProblemeAResoudre->X[Var];
+    }
+}
+
+void CSR_DEBUG_BLAH(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre)
+{
+    int Var;
+
+    logs.info();
+    logs.info() << LOG_UI_DISPLAY_MESSAGES_OFF;
+    logs.info() << "Here is the trace:";
+
+    for (Var = 0; Var < ProblemeAResoudre->NombreDeVariables; Var++)
+    {
+        logs.info().appendFormat("Variable %ld cout lineaire %e  cout quadratique %e",
+                                 Var,
+                                 ProblemeAResoudre->CoutLineaire[Var],
+                                 ProblemeAResoudre->CoutQuadratique[Var]);
+    }
+    for (int Cnt = 0; Cnt < ProblemeAResoudre->NombreDeContraintes; Cnt++)
+    {
+        logs.info().appendFormat("Constraint %ld sens %c B %e",
+                                 Cnt,
+                                 ProblemeAResoudre->Sens[Cnt],
+                                 ProblemeAResoudre->SecondMembre[Cnt]);
+
+        int il = ProblemeAResoudre->IndicesDebutDeLigne[Cnt];
+        int ilMax = il + ProblemeAResoudre->NombreDeTermesDesLignes[Cnt];
+        for (; il < ilMax; ++il)
+        {
+            Var = ProblemeAResoudre->IndicesColonnes[il];
+            logs.info().appendFormat("      coeff %e var %ld xmin %e xmax %e type %ld",
+                                     ProblemeAResoudre->CoefficientsDeLaMatriceDesContraintes[il],
+                                     Var,
+                                     ProblemeAResoudre->Xmin[Var],
+                                     ProblemeAResoudre->Xmax[Var],
+                                     ProblemeAResoudre->TypeDeVariable[Var]);
+        }
+    }
+}
+
+void handleInteriorPointError(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre,
+                              HOURLY_CSR_PROBLEM& hourlyCsrProblem,
+                              uint weekNb,
+                              int yearNb)
+{
+    const int hoursInWeek = 168;
+    logs.warning()
+      << "No further optimization for CSR is possible, optimum solution is set as LMR . year: "
+      << yearNb + 1
+      << ". hour: " << weekNb * hoursInWeek + hourlyCsrProblem.hourInWeekTriggeredCsr + 1;
+
+#ifndef NDEBUG
+    CSR_DEBUG_BLAH(ProblemeAResoudre);
+#endif
+}
+
+bool ADQ_PATCH_CSR(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre,
+                   HOURLY_CSR_PROBLEM& hourlyCsrProblem,
+                   uint weekNb,
+                   int yearNb)
+{
+    PROBLEME_POINT_INTERIEUR Probleme;
+    setInteriorPointProblem(ProblemeAResoudre, Probleme);
+    PI_Quamin(&Probleme); // resolution
```

Comment:
Slightly clearer version
```suggestion
    std::unique_ptr<PROBLEME_POINT_INTERIEUR> Probleme = buildInteriorPointProblem(ProblemeAResoudre);
    PI_Quamin(Probleme.get()); // resolution
```